src/visuals.js

```diff
@@ -383,26 +405,45 @@ export const Visuals = {
     // Calculate direction from bottom left to mouse
     let dirX = scaleX(this.p.mouseX) - startX
     let dirY = scaleY(this.p.mouseY) - startY
+    this.p.line(
+      scaleX(this.p.mouseX) - crossHairSize,
+      scaleX(this.p.mouseY),
+      scaleX(this.p.mouseX) + crossHairSize,
+      scaleX(this.p.mouseY)
+    )
+    this.p.line(
+      scaleX(this.p.mouseX),
+      scaleX(this.p.mouseY) - crossHairSize,
+      scaleX(this.p.mouseX),
+      scaleX(this.p.mouseY) + crossHairSize
+    )
+    // // Calculate the length of the direction
+    // let len = this.p.sqrt(dirX * dirX + dirY * dirY)
 
-    // Calculate the length of the direction
-    let len = this.p.sqrt(dirX * dirX + dirY * dirY)
-
-    // If the length is not zero, scale the direction to have a length of 100
-    if (len != 0) {
-      dirX = (dirX / len) * 100
-      dirY = (dirY / len) * 100
-    }
+    // // If the length is not zero, scale the direction to have a length of 100
+    // if (len != 0) {
+    //   dirX = (dirX / len) * 100
+    //   dirY = (dirY / len) * 100
+    // }
 
     // Draw the line
+    // this.p.setLineDash([5, 15])
+    const drawingContext = this.p.canvas.getContext('2d')
+    const chunk = this.windowWidth / 100
+    drawingContext.setLineDash([chunk])
+    drawingContext.lineDashOffset = -(this.frames * 10)
```

Comment:
What's the animation that's supposed to happen here with the dashed line? What I see is it shakes then stops then shakes then stops but timing doesn't seem consistent.

I like the line idea but find the animation a little confusing. Maybe kill it or make it slower by removing the `* 10`?

Reply:
ah interesting, you use safari right? maybe looks different there. p5 doesn't have any dashed line capability but native canvas context has it so I add line dash of chunk size, then I offset it based on the frame number so that it's a moving dashed line. I set it to 10 because then it matches the speed of the missile as it leaves the gun. Actually helped match up speed and direction on my machine but that's when it was working correctly. I'll take a look on Safari. Wanted to disable by default anyway tho and make it possibly like a paid option. Can add that here already.